PolyEngine/Engine/CMakeLists.txt

```diff
@@ -0,0 +1,61 @@
+find_package(OpenGL REQUIRED)
+if(WIN32)
+	find_package(GLEW REQUIRED PATHS ThirdParty/glew/lib/Release/ PATH_SUFFIXES Win32 x64) #FIXME(vuko): does this even work? how to make it search default paths first? (according to docs it should by default, but it fails on my system as opposed to invocation without given paths)
+else(WIN32)
+	find_package(GLEW REQUIRED)
+endif(WIN32)
+
+
+set(POLYENGINE_SRCS
+	Src/Engine.cpp
+	Src/Entity.cpp
+	Src/OpenGLRenderingContext.cpp
+	Src/World.cpp
+	Src/CameraComponent.cpp
+	Src/CameraSystem.cpp
+	Src/InputSystem.cpp
+	Src/InputWorldComponent.cpp
+	Src/MeshRenderingComponent.cpp
+	Src/RenderingSystem.cpp
+	Src/ShaderProgram.cpp
+	Src/TransformComponent.cpp
+	Src/ViewportWorldComponent.cpp
+)
+set(POLYENGINE_INCLUDE
+	Src
+)
+set(POLYENGINE_H_FOR_IDE
+	Src/CameraComponent.hpp
```

Comment:
it will regenerate everything, i've used it sucessfully, read example attached in previous comment.